tests/queries/0_stateless/01917_distinct_on.sql

```diff
@@ -0,0 +1,9 @@
+DROP TABLE IF EXISTS t1;
+
+CREATE TABLE t1 (`a` UInt32, `b` UInt32, `c` UInt32 ) ENGINE = Memory;
+INSERT INTO t1 VALUES (1, 1, 1), (1, 1, 2), (2, 2, 2), (1, 2, 2);
+
+SELECT DISTINCT ON (a, b) a, b, c FROM t1;
```

Comment:
Also it's good to add test to `DISTINCT ON` and `LIMIT BY` to check that correct error returned. To check error code add `-- { serverError <exprecred_code> }` like this

https://github.com/ClickHouse/ClickHouse/blob/9f5cd35a6963cc556a51218b46b0754dcac7306a/tests/queries/0_stateless/01601_accurate_cast.sql#L1

Reply:
I have added a test.

Reply:
But the test failed which returned `Code: 590. DB::Exception: Can not use DISTINCT ON alongside LIMIT BY: 
Expected server error '590'.`

Reply:
Exception on server returned as server error (if it's not `LOGICAL_ERROR`). Failed test case looks weird I'll check it.

Reply:
Because it's `clientError`, not `serverError`